### DISCUSSION TITLE
How to handle temporary incoming folder with external library

### DISCUSSION THREAD

**Discussion by AlmightyFrog** [immich-app/immich#14810] [Q&A]
Hi,
i am using Immich since quite some time, had once a broken installation and needed to regenerate all thumbnails/videos again.
What I still not am sure about is how my use-case can be best handled. So far I do only use external libraries and do not intent do change that to the inbuilt upload feature for current devices:

1. Smartphones (3 devices -> 5 folders) sync with syncthing to let's say /media/sync/{devicename} immediately using syncthing.
This is my "live" view of what happened lately and what i really like. Pictures stay on Smartphones. Some of them might be deleted as they're duplicate/not relevant.

2. Every few months (ideally) i move and sort all pictures to different structure /media/photos/{source}/{year}/{quarter} which results in:
     - they get removed automatically from smartphones (syncthing)
     - immich detects them as new images and starts all the thumbnail/transcoding again. also loses all detected faces and so on.

That is not ideal. What can I do about it? I already tried using separate external libraries for /media/sync and /media/photos but that does not really help either.

For photos made with my classic camera there is not that huge issue, as i directly copy them manually from SD-Card to /media/photos/ and there is no dynamic syncing.

Any suggestions are welcome (also others including maybe changing workflow but syncthing is a given).

**Comment by archont00** [comment]
I have exactly the same workflow with FolderSync (instead of Syncthing) and agree it would be nice if Immich detected that an existing photo file changed location rather than consider it removal and a new addition.

I think there's a similar discussion topic here: https://github.com/immich-app/immich/discussions/3533

**Comment by gschintgen** [comment]
I'm using your workflow too: upload library for everyday use, immediate availability for sharing but periodically I'll import the pictures in Digikam to sort, tag, etc. and then they're added back in Immich as part of a large (read-only) external library.

I'm not so much bothered by the regeneration of thumbnails and such but what is bothering me is that I have to manually delete the originals. And that operation is a bit cumbersome because I can't just temporarily hide my external library.

**Comment by Shurov** [comment]
Second this. 

I have a bit simpler approach, that also lacks Immich solution to workflow, so planning to using smth like syncthing...

My current (problematic and lacking) approach without external synchroniser, pure Immich...:
Obviously, I have huge archive of old photos that I also don't want to move into native Immich library - so they stay within External Library.
I'd like to use Immich native photos upload function, but they are uploaded into Immich Upload location. It's ok. But once in a while I want to move those photos into the main folder with photos (which is imported as an External Lib). Once I move the photos between 2 folders (so, outside Immich control), it just loses all those files, and constantly shows them as missing. So that Timeline is full of errors...

Ideally, I'd like to marry Immich Upload with an external folder. However, currently, it's unfeasible as it seems. Hence, I thought I should use 3rd party solution like syncthing to upload mobile photos on the go, so that Immich wouldn't have them in its native library...
